host/lib/usrp/dboard/magnesium/magnesium_radio_control.cpp

```diff
@@ -1018,6 +1020,39 @@ bool magnesium_radio_control_impl::get_lo_lock_status(const direction_t dir)
     return lo_lock;
 }
 
+double magnesium_radio_control_impl::_predict_actual_lo_freq(double ret_freq){
+    // The frequency returned here is a lie. MYKONOS_setRfPllFrequency accepts integers only
+    // I'm doing my best here to get the actual frequency based on the info on
+    // AD9371-9375-User-Guide-UG-992.pdf and experimental data
+    //
+    // Depending on the frequency range, ad9371 adopts a set denominator
+    // for the fractional N PLL:
+    // From 400MHz to 750MHz   : divide by 16
+    // From 750MHz to 1500MHz  : divide by 8
+    // From 1500MHz to 3000MHz : divide by 4
+    // From 3000MHz to 6000MHz : divide by 2
+    // This table comes straight from the user guide, from section
+    // RF PLL FREQUENCY CHANGE PROCEDURE.
+    // Now, from the the table in section RF PLL RESOLUTION LIMITATIONS,
+    // we have some frequency steps depending on the master clock rate
+    // Dividing the master clock rate by the frequency steps, and combining
+    // with some experimental data analyzing the possible frequency steps,
+    // we arrive at the formula for frequency_step below
+    // Any requested frequency will be truncated to this step.
+    unsigned denominator = 10000;
+    if (ret_freq > 400e6 && ret_freq <= 750e6){
+        denominator = 16;
+    }else if(ret_freq > 750e6 && ret_freq <= 1500e6){
+        denominator = 8;
+    }else if(ret_freq > 1500e6 && ret_freq <= 3000e6){
+        denominator = 4;
+    }else if(ret_freq > 3000e6 && ret_freq <= 6000e6){
+        denominator = 2;
+    }
+    double frequency_step = (double)2 * _master_clock_rate / 33546240.0 / denominator;
```

Comment:
Where did you find the value 33546240?

Reply:
...sorry, you explain that in the comment above. Very good.

Reply:
Yeah, it's 100% experimental unfortunately, I tuned the frequencies from n300, and watched what the spectrum analyzer reported. Did the reverse math for the first few, this value came up, and it held up after a bunch of other frequencies.

There seems to be a better documentation for this(I forgot the reference) but this must be behind some kind of NDA, because I couldn't find it anywhere, I think they just expect you to use mykonos anyway.

Reply:
Well, they expect you to use Mykonos, but then also read UG-992 and do the math as they explain. So I think this is actually the "right way" to do things... even if it's a bit weird. It's annoying that Table 97 doesn't have more rates available (like our 125e6 rate).